test/e2e/reconciler/locationworkspace/local_apiexport_test.go

```diff
@@ -0,0 +1,188 @@
+	syncTarget := framework.NewSyncerFixture(t, source, computeClusterName,
+		framework.WithAPIExports(""),
+		framework.WithExtraResources("services"),
+		framework.WithSyncTarget(computeClusterName, syncTargetName),
+		framework.WithDownstreamPreparation(func(config *rest.Config, isFakePCluster bool) {
+			if !isFakePCluster {
+				// Only need to install services
+				return
+			}
+			sinkCrdClient, err := apiextensionsclientset.NewForConfig(config)
+			require.NoError(t, err, "failed to create apiextensions client")
+			t.Logf("Installing test CRDs into sink cluster...")
+			kubefixtures.Create(t, sinkCrdClient.ApiextensionsV1().CustomResourceDefinitions(),
+				metav1.GroupResource{Group: "core.k8s.io", Resource: "services"},
+			)
+			require.NoError(t, err)
+		}),
+	).Start(t)
+
+	require.Eventually(t, func() bool {
+		syncTarget, err := kcpClients.Cluster(computeClusterName).WorkloadV1alpha1().SyncTargets().Get(ctx, syncTargetName, metav1.GetOptions{})
+		if err != nil {
+			return false
+		}
+
+		if len(syncTarget.Status.SyncedResources) != 1 {
+			return false
+		}
+
+		if syncTarget.Status.SyncedResources[0].Resource != "services" ||
+			syncTarget.Status.SyncedResources[0].State != workloadv1alpha1.ResourceSchemaAcceptedState {
+			return false
+		}
+
+		return true
+	}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+	// create virtual workspace rest configs
+	rawConfig, err := source.RawConfig()
+	require.NoError(t, err)
+	virtualWorkspaceRawConfig := rawConfig.DeepCopy()
+	virtualWorkspaceRawConfig.Clusters["syncvervw"] = rawConfig.Clusters["base"].DeepCopy()
+	virtualWorkspaceRawConfig.Clusters["syncvervw"].Server = rawConfig.Clusters["base"].Server + "/services/syncer/" + computeClusterName.String() + "/" + syncTargetName + "/" + syncTarget.SyncerConfig.SyncTargetUID
+	virtualWorkspaceRawConfig.Contexts["syncvervw"] = rawConfig.Contexts["base"].DeepCopy()
+	virtualWorkspaceRawConfig.Contexts["syncvervw"].Cluster = "syncvervw"
+	virtualWorkspaceConfig, err := clientcmd.NewNonInteractiveClientConfig(*virtualWorkspaceRawConfig, "syncvervw", nil, nil).ClientConfig()
+	require.NoError(t, err)
+	virtualWorkspaceConfig = kcpclienthelper.SetMultiClusterRoundTripper(rest.AddUserAgent(rest.CopyConfig(virtualWorkspaceConfig), t.Name()))
+
+	virtualWorkspaceiscoverClusterClient, err := clientgodiscovery.NewDiscoveryClientForConfig(virtualWorkspaceConfig)
+	require.NoError(t, err)
+
+	require.Eventually(t, func() bool {
+		_, existingAPIResourceLists, err := virtualWorkspaceiscoverClusterClient.WithCluster(logicalcluster.Wildcard).ServerGroupsAndResources()
+		if err != nil {
+			return false
+		}
+		// requiredAPIResourceList includes all core APIs plus services API
+		return len(cmp.Diff([]*metav1.APIResourceList{
+			requiredAPIResourceListWithService(computeClusterName, computeClusterName)}, sortAPIResourceList(existingAPIResourceLists))) == 0
+	}, wait.ForeverTestTimeout, time.Millisecond*100)
```

Comment:
Could you add a message in the require.Eventually, to explain what was expected and not met (Services API being available in the Virtual Workspace) ?

Reply:
added